File: sdk-swift/Sources/DecibelSDK/Models/Order.swift
import Foundation

public struct UserOpenOrder: Codable {
    public let market: String
    public let orderID: String
    public let clientOrderID: String?
    public let price: Double
    public let origSize: Double
    public let remainingSize: Double
    public let isBuy: Bool
    public let timeInForce: String
    public let isReduceOnly: Bool
    public let status: String
    public let transactionUnixMs: Int64
    public let transactionVersion: Int64

    enum CodingKeys: String, CodingKey {
        case market
        case orderID = "order_id"
        case clientOrderID = "client_order_id"
        case price
        case origSize = "orig_size"
        case remainingSize = "remaining_size"
        case isBuy = "is_buy"
        case timeInForce = "time_in_force"
        case isReduceOnly = "is_reduce_only"
        case status
        case transactionUnixMs = "transaction_unix_ms"
        case transactionVersion = "transaction_version"
    }
}

public struct OrderStatus: Codable {
    public let parent: String
    public let market: String
    public let orderID: String
    public let status: String
    public let origSize: Double
    public let remainingSize: Double
    public let sizeDelta: Double
    public let price: Double
    public let isBuy: Bool
    public let details: String
    public let transactionVersion: Int64
    public let unixMs: Int64

    enum CodingKeys: String, CodingKey {
        case parent, market
        case orderID = "order_id"
        case status
        case origSize = "orig_size"
        case remainingSize = "remaining_size"
        case sizeDelta = "size_delta"
        case price
        case isBuy = "is_buy"
        case details
        case transactionVersion = "transaction_version"
        case unixMs = "unix_ms"
    }
}

public struct UserActiveTwap: Codable {
    public let market: String
    public let isBuy: Bool
    public let orderID: String
    public let clientOrderID: String
    public let isReduceOnly: Bool
    public let startUnixMs: Int64
    public let frequencyS: Int64
    public let durationS: Int64
    public let origSize: Double
    public let remainingSize: Double
    public let status: String
    public let transactionUnixMs: Int64
    public let transactionVersion: Int64

    enum CodingKeys: String, CodingKey {
        case market
        case isBuy = "is_buy"
        case orderID = "order_id"
        case clientOrderID = "client_order_id"
        case isReduceOnly = "is_reduce_only"
        case startUnixMs = "start_unix_ms"
        case frequencyS = "frequency_s"
        case durationS = "duration_s"
        case origSize = "orig_size"
        case remainingSize = "remaining_size"
        case status
        case transactionUnixMs = "transaction_unix_ms"
        case transactionVersion = "transaction_version"
    }
}

File: sdk-go/models/order.go
package models

// UserOpenOrder represents an active open order.
type UserOpenOrder struct {
	Market             string  `json:"market"`
	OrderID            string  `json:"order_id"`
	ClientOrderID      *string `json:"client_order_id"`
	Price              float64 `json:"price"`
	OrigSize           float64 `json:"orig_size"`
	RemainingSize      float64 `json:"remaining_size"`
	IsBuy              bool    `json:"is_buy"`
	TimeInForce        string  `json:"time_in_force"`
	IsReduceOnly       bool    `json:"is_reduce_only"`
	Status             string  `json:"status"`
	TransactionUnixMs  int64   `json:"transaction_unix_ms"`
	TransactionVersion int64   `json:"transaction_version"`
}

// UserOrderHistoryItem represents an order in history.
type UserOrderHistoryItem struct {
	Market             string  `json:"market"`
	OrderID            string  `json:"order_id"`
	ClientOrderID      *string `json:"client_order_id"`
	Price              float64 `json:"price"`
	OrigSize           float64 `json:"orig_size"`
	RemainingSize      float64 `json:"remaining_size"`
	IsBuy              bool    `json:"is_buy"`
	TimeInForce        string  `json:"time_in_force"`
	IsReduceOnly       bool    `json:"is_reduce_only"`
	Status             string  `json:"status"`
	TransactionUnixMs  int64   `json:"transaction_unix_ms"`
	TransactionVersion int64   `json:"transaction_version"`
}

// OrderStatus represents the status of an order.
type OrderStatus struct {
	Parent             string  `json:"parent"`
	Market             string  `json:"market"`
	OrderID            string  `json:"order_id"`
	Status             string  `json:"status"`
	OrigSize           float64 `json:"orig_size"`
	RemainingSize      float64 `json:"remaining_size"`
	SizeDelta          float64 `json:"size_delta"`
	Price              float64 `json:"price"`
	IsBuy              bool    `json:"is_buy"`
	Details            string  `json:"details"`
	TransactionVersion int64   `json:"transaction_version"`
	UnixMs             int64   `json:"unix_ms"`
}

// UserActiveTwap represents an active TWAP order.
type UserActiveTwap struct {
	Market             string  `json:"market"`
	IsBuy              bool    `json:"is_buy"`
	OrderID            string  `json:"order_id"`
	ClientOrderID      string  `json:"client_order_id"`
	IsReduceOnly       bool    `json:"is_reduce_only"`
	StartUnixMs        int64   `json:"start_unix_ms"`
	FrequencyS         int64   `json:"frequency_s"`
	DurationS          int64   `json:"duration_s"`
	OrigSize           float64 `json:"orig_size"`
	RemainingSize      float64 `json:"remaining_size"`
	Status             string  `json:"status"`
	TransactionUnixMs  int64   `json:"transaction_unix_ms"`
	TransactionVersion int64   `json:"transaction_version"`
}
